### ISSUE TITLE
Can't figure out how to use nGlide in window mode properly

### ISSUE THREAD

**Issue by User-3090** [joncampbell123/dosbox-x#3893]
### Question

I want to launch 3DFX games in Dosbox-X in Window Mode with fixed/specified dimensions, but I'm not smart enough to figure it out.

- I realized Alt+Enter make nGlide go to Window Mode
- That Window Mode is still not resizable and partly it's behind the Windows task bar. But aspect ration (4:3) is preserved.
- When in Window Mode it still seems Dosbox-X thinks it's in full screen mode. Pressing F11+f then it seems to go to a random Window size and messes up aspect ratio. The window is still not resizable
- I figured that setting Dosbox-X to Direct3D output makes the Window technically resizable using some Autohotkey hackery

- Is there really no way to launch a game without going to fullscreen mode first?
- Is there really no way I can launch a game in Window mode with fixed/specified dimensions?
- Is there really no way to have a resizable Window?

Also the cursor is always hidden when over the nGlide Window and it's always above all other Windows. Is there a way to change this?

I'm using the SDL2 build, as with SDL1 Alt-Enter in nGlide does nothing for me. Is this a problem with nGlide only? Will other glide wrappers allow me to play in Window mode properly?

![2022-12-21 11_50_53-DOSBox-X 2022 09 0_ S2_3DFX - 100%](https://user-images.githubusercontent.com/71881005/208889964-7dcd157a-045e-42eb-8ed3-48907defac33.png)
![2022-12-21 11_51_14-](https://user-images.githubusercontent.com/71881005/208890035-1e1a7c09-f3e5-420f-91bc-1dc661d422c4.png)
![2022-12-21 11_52_22-DOSBox-X 2022 09 0_ S2_3DFX - 100%](https://user-images.githubusercontent.com/71881005/208890095-4a1256b9-d5d7-459b-8a00-3ad94fbb90c1.png)

### Have you checked that no similar question(s) exist?

- [X] I have searched and didn't find any similar question.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by User-3090** [comment]
I'm using the 32bit version and apart from the inability to resize or scale the new windows nGlide is working find. I'll give dgVoodoo2 a spin. Thanks!

**Comment by User-3090** [comment]
Oh man. It all magically works when using dgVoodo2. Can't believe I spend so much time getting things to work with nGlide. Are there particular settings I should use with Dosbox-x to optimize for dgVoodo2? I assume set at least output to D3D? Thanks!!!

**Comment by User-3090** [comment]
It mostly works. However I get artifacts in menus as soon as I increase the resolution in the dgVoodoo2 control panel. Any idea what might be causing that? I tried all options I could think of.

Thanks!

![image](https://user-images.githubusercontent.com/71881005/208936466-55889824-1598-4cae-8d54-8f10f7aae500.png)

**Comment by kjliew** [comment]
> * Is there really no way I can launch a game in Window mode with fixed/specified dimensions?

No, you can. If you haven't watched the video below, then you should take a peek. 😜
https://www.youtube.com/watch?v=QByfzl8Ij3Y&t=720s

dgVoodoo2 and nGlide are great when OpenGLide lacks the compatibility quirks to manage a handful of tricky Glide games. But when OpenGLide works (Screamer 2 / Rally are great examples), its openness enables seamless integration with the host application without having to deal with separate wrappers configuration.

**Comment by User-3090** [comment]
> > * Is there really no way I can launch a game in Window mode with fixed/specified dimensions?
> 
> No, you can. If you haven't watched the video below, then you should take a peek. 😜 https://www.youtube.com/watch?v=QByfzl8Ij3Y&t=720s
> 
That video was the reason i just donated to your project ;) looking forward to try it.

**Comment by kjliew** [comment]
Oh, Thank YOU very, very much!!!👼
Let's me know if you wish to bundle the package to include DOSBox SVN with Games essential patches that I used in the video.  It is now a much improved DOSBox SVN with `sdl12-compat` out-of-the-box, seamless scaling regardless of Glide pass-through or VOODOO emulation. DOSBox is always better for DOS games and a worthy complement to QEMU for retro gaming.

I hope DOSBox-X wouldn't mind such promotion here. 😜

**Comment by JayQ2K** [comment]
> Oh man. It all magically works when using dgVoodo2. Can't believe I spend so much time getting things to work with nGlide. Are there particular settings I should use with Dosbox-x to optimize for dgVoodo2? I assume set at least output to D3D? Thanks!!!

Not really. I pretty much use it out of the box with just glide=true. You can take a look at the configuration panel application that comes shipped with dgVoodoo2 and see what can help there. Haven't used it myself combined with Dosbox so can't say what would work (if at all, because Dosbox might disregard those settings).
I work with a mini-config file myself that only changes settings that are not default and run that file from a shortcut like:
D:\DOSBox-X\dosbox-x.exe -conf gamebox.conf

gamebox.conf is filled with:
[voodoo]
glide                   = true

[config]
break       = off
dos         = high, umb
fcbs        = 4
files       = 30
lastdrive   = a
and some other stuff, but not the whole config file. This keeps things in order (for me at least) and only when required to add settings I will bring them over from the default config file and adjust them here.

**Comment by JayQ2K** [comment]
> I hope DOSBox-X wouldn't mind such promotion here. 😜

Well, your project is mentioned [here](https://dosbox-x.com/wiki/Guide%3ASetting-up-3dfx-Voodoo-in-DOSBox%E2%80%90X#_windows_host_installing_glide2x_dll), so I guess not.

**Comment by User-3090** [comment]
Basically all the scaling options in Dosbox-x as well as in the dgVoodoo2 control panel are broken or have no effect. I tried each and every setting and combination of builds. The best I can get is SDL2 build with dgVoodoo2 set to 2x on Glide tab. But even that is quirky. Minimize and restore the Dosbox-x window and it's messed up again. 

![image](https://user-images.githubusercontent.com/71881005/209173393-155542b2-e2d9-45a2-aa6b-398cc870d5c3.png)

**Comment by User-3090** [comment]
Also Direct-X shaders do not work at all with dgVoodo2. Those worked with nGlide at least.

Window mode in Dosbox-x with nGlide and dgVoodo2 all seems very quirky and mostly broken to me. I'm mean on the positive side it there is a picture and it does render, but that's about it ;)

**Comment by JayQ2K** [comment]
In that case I'm gonna look into @kjliew project and see what happens there.
Glide and OpenGL are more native to each other than Glide and D3D (though as we can see it does work).

You might want to check if output=opengl does the trick.

**Comment by kjliew** [comment]
> In that case I'm gonna look into @kjliew project and see what happens there.

What happens here is REALLY simple -- I **\*LOVE\*** playing my 3Dfx games on DOSBox -- to the point that it offended VOGONS Marvin folks for their troubles with 3Dfx games (Carmageddon 3Dfx, Tomb Raider 3Dfx, etc.) on real junkyard PCs with "whatever-Voodoo". The Glide pass-through & VOODOO emulation integration aren't just another checkbox syndrome. Seamless scaling, full-screen and "real" 3D accelerated rendering are priority for good game experience on modern systems.

In fact, there was numerous accounts that the main devs acknowledged that they didn't have that much focus on games. Perhaps the project is better funded by helping small businesses/offices to protect their investment in DOS software for word processing/accounting/inventory tracking/invoice printing etc.

**Comment by JayQ2K** [comment]
You don't deliver binaries for those that just want to play games instead of building the files themselves?

**Comment by kjliew** [comment]
> You don't deliver binaries for those that just want to play games instead of building the files themselves?

I do now as the bundle request for those donated to my project for QEMU featuring qemu-3dfx. 😁 I cordially call it **\*the Ultimate Retro PC Gaming pack\*** on modern systems for thousands of games from the DOS era into the 1996-2004 Windows gaming "donut hole".

DOSBox has never been short of community builds. The patches are GPL'ed and available for everyone.
